lib/dry/schema/key_validator.rb

```diff
@@ -51,9 +51,14 @@ def key_paths(hash)
         hash.flat_map { |key, _|
           case (value = hash[key])
           when Hash
+            next key.to_s if value.empty?
+
             [key].product(key_paths(hash[key])).map { |keys| keys.join(DOT) }
           when Array
-            hashes_or_arrays = value.select { |e| e.is_a?(Array) || e.is_a?(Hash) }
+            hashes_or_arrays = value.select { |e| (e.is_a?(Array) || e.is_a?(Hash)) && !e.empty? }
+
+            next key.to_s if hashes_or_arrays.empty?
+
```

Comment:
since it's skipping empy items now, aren't we going to end up with `nil`s in the resulting array?

Reply:
The `next` keyword takes one argument which will be taken by the iterator. In this case that would be `key.to_s`. 🙂 

```ruby
3.times.map { |n| next n }
#=> [0, 1, 2]
```

Reply:
Ah doooh of course! Thanks for the explanation 😄